### PR TITLE
Add User List API functionality

### PR DESCRIPTION
**Pull request by acher82**

### CODE REVIEW

File: conjur-api/User.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// <copyright file="User.cs" company="Conjur Inc.">
+//     Copyright (c) 2016-2018 Conjur Inc. All rights reserved.
```
Comment (dividedmind) [inline conjur-api/User.cs]: Since it's a new file, the copyright should probably be 2018 Cyberark Ltd.

File: conjur-api/User.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace Conjur
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Conjur User reference.
+    /// </summary>
+    /// A user represents an identity for a human. It is a role, in RBAC terms.
```
Comment (dividedmind) [inline conjur-api/User.cs]: Nitpick: strictly speaking it's both a role and a resource, and here you're referencing the resource part.

File: conjur-api/User.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public class User : Resource
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Conjur.User"/> class.
+        /// </summary>
+        /// <param name="client">Conjur client to use to connect.</param>
+        /// <param name="name">The User (or host) name.</param>
```
Comment (dividedmind) [inline conjur-api/User.cs]: Host won't work here as far as I can tell.

File: conjur-api/User.cs (hunk 1 above)
Comment (DvirCyberArk) [reply conjur-api/User.cs]: done

File: conjur-api/User.cs (hunk 3 above)
Comment (DvirCyberArk) [reply conjur-api/User.cs]: removed (or host)

File: conjur-api/User.cs (hunk 2 above)
Comment (DvirCyberArk) [reply conjur-api/User.cs]: Change description to describe user as resource only

**Pull request merged**